src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -293,10 +293,12 @@ def _parse_file_name(self, path):
         stream = get_entry(matched, "scan_id")
         file_idx = int(get_entry(matched, "file_idx_in_scan"))
         try:
-            idx_offset = get_entry(matched, "file_idx_offset")
+            idx_offset = int(get_entry(matched, "file_idx_offset"))
             file_idx += idx_offset
         except utils.UsageError:
             pass
+        except ValueError:
+            logger.warning("Could not read `file_idx_offset` from %s. Offset ignored.", path)
```

Comment:
I am not sure that ignoring is a good idea. Would it be better to ingest a file to the wrong index (and possibly stream) instead of not ingesting it at all? For example, a missing entry will likely be noticed early, but ingesting to the wrong index might stay unnoticed for much longer.

In any case, this needs a comment documenting the reasoning for the chosen behavior.